sycl/include/CL/sycl/atomic.hpp

```diff
@@ -274,12 +280,20 @@ class atomic {
 #endif
   }
 
-  T fetch_add(T Operand, memory_order Order = memory_order::relaxed) {
-    __SYCL_STATIC_ASSERT_NOT_FLOAT(T);
+  template <typename T2 = T>
+  detail::enable_if_t<std::is_integral<T2>::value, T>
+  fetch_add(T Operand, memory_order Order = memory_order::relaxed) {
     return __spirv_AtomicIAdd(
         Ptr, SpirvScope, detail::getSPIRVMemorySemanticsMask(Order), Operand);
   }
 
+  template <typename T2 = T>
+  detail::enable_if_t<std::is_floating_point<T2>::value, T>
+  fetch_add(T Operand, memory_order Order = memory_order::relaxed) {
+    return __spirv_AtomicFAdd(
+        Ptr, SpirvScope, detail::getSPIRVMemorySemanticsMask(Order), Operand);
+  }
+
   T fetch_sub(T Operand, memory_order Order = memory_order::relaxed) {
```

Comment:
Should you make the same change here as for `fetch_add`?  It might be a little surprising that we support `fetch_add` and `fetch_sub` in `atomic_ref`, but only `fetch_add` for `atomic`.

Reply:
Done; should be aligned properly now.